compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java

```diff
@@ -998,23 +997,52 @@ private boolean isBBCoveredInErrorEntry(BIRErrorEntry errorEntry, List<BIRBasicB
      *
-     * @param pos The position for which the ID should be generated
+     * @param pkg The module for which the ID should be generated
      * @return The generated ID
      */
-    private String generatePositionId(Location pos) {
-        return String.format("%s:%d:%d", pos.lineRange().filePath(), pos.lineRange().startLine().line() + 1,
-                pos.lineRange().startLine().offset() + 1);
+    private String generatePackageId(PackageID pkg) {
+        return String.format("%s/%s:%s", pkg.orgName.value, pkg.name.value, getMajorVersion(pkg.version.value));
     }
 
     /**
-     * Generate a ID for a ballerina module.
+     * Generate operands for location.
      *
-     * @param pkg The module for which the ID should be generated
-     * @return The generated ID
+     * @param pkg       Bir package
+     * @param func      Bir Function
+     * @param currentBB Current basic block
+     * @param location  Location
+     * @return List of operands for source file name, start and end positions
      */
-    private String generatePackageId(PackageID pkg) {
-        return String.format("%s/%s:%s", pkg.orgName.value, pkg.name.value, getMajorVersion(pkg.version.value));
+    private List<BIROperand> generatePositionArgs(BIRPackage pkg, BIRFunction func, BIRBasicBlock currentBB,
+                                                  Location location) {
+        String pkgId = generatePackageId(pkg.packageID);
+        BIROperand pkgOperand = generateGlobalConstantOperand(pkg, symbolTable.stringType, pkgId);
+        Name fileName = new Name("$observabilityFileName" + localVarIndex++);
+        Name startPosition = new Name("$observabilityStart" + localVarIndex++);
+        Name endPosition = new Name("$observabilityEnd" + localVarIndex++);
+        BIRVariableDcl positionFileVar = new BIRVariableDcl(symbolTable.stringType, fileName, VarScope.FUNCTION,
+                VarKind.TEMP);
+        BIRVariableDcl positionStartVar = new BIRVariableDcl(symbolTable.intType, startPosition, VarScope.FUNCTION,
+                VarKind.TEMP);
+        BIRVariableDcl positionEndVar = new BIRVariableDcl(symbolTable.intType, endPosition, VarScope.FUNCTION,
+                VarKind.TEMP);
+        BIROperand fileNameOperand = new BIROperand(positionFileVar);
+        BIROperand startPositionOperand = new BIROperand(positionStartVar);
+        BIROperand endPositionOperand = new BIROperand(positionEndVar);
+        ConstantLoad fileNameIns = new ConstantLoad(location, location.lineRange().filePath(),
+                fileNameOperand.variableDcl.type, fileNameOperand);
+        ConstantLoad startPositionIns = new ConstantLoad(location, location.lineRange().startLine().line() + 1,
+                startPositionOperand.variableDcl.type, startPositionOperand);
+        ConstantLoad endPositionIns = new ConstantLoad(location, location.lineRange().startLine().offset() + 1,
+                endPositionOperand.variableDcl.type, endPositionOperand);
```

Comment:
IINM, this is the `startLine` and `startCol` isn't it ?
- location.lineRange().startLine().line() - start line
- location.lineRange().startLine().offset() - start column

So shall we update it ? (I think it's there in other places as well)

Reply:
Fixed